backend/tests/utils/graph_db_utils_test.py

```diff
@@ -32,11 +37,52 @@ def test_database_connectivity_is_healthy(mock_driver):
     mock_driver.close.assert_called_once()
 
 
-def test_database_connectivity_is_unhealthy(mock_driver):
+def test_database_connectivity_is_unhealthy(mocker, mock_driver):
+    mocker.patch("src.utils.graph_db_utils.driver", mock_driver)
     mock_driver.verify_connectivity.side_effect = Exception
 
     connected = verify_connection()
 
     assert not connected
     mock_driver.verify_connectivity.assert_called_once()
     mock_driver.close.assert_called_once()
+
+
+def test_populate_db_populates_db(mocker, mock_driver, mock_session):
+    mocker.patch("src.utils.graph_db_utils.driver", mock_driver)
+
+    query = "CREATE (n:Test {data: $all_data})"
+    data = {"key": "value"}
+    remove_credits = "REMOVE CREDITS"
+    remove_transactions_without_merchant = "REMOVE TRANSACTIONS"
+    remove_connecting_nodes = "REMOVE NODES"
+
+    mocker.patch("src.utils.graph_db_utils.remove_credits", remove_credits)
+    mocker.patch("src.utils.graph_db_utils.remove_transactions_without_merchant", remove_transactions_without_merchant)
+    mocker.patch("src.utils.graph_db_utils.remove_connecting_nodes", remove_connecting_nodes)
+
+    populate_db(query, data)
+
+    mock_session.run.assert_any_call("MATCH (n) DETACH DELETE n")
+    mock_session.run.assert_any_call(query, data={"all_data": data})
+    mock_session.run.assert_any_call(remove_credits)
+    mock_session.run.assert_any_call(remove_transactions_without_merchant)
+    mock_session.run.assert_any_call(remove_connecting_nodes)
```

Comment:
I don't know what I prefer, so up to you, but you could get the order asserted here too (assuming that `assert_any_call` just looks for any time that the method was called with) with something like in meesage_handlers_test:
```python
first_call = mocked_create_task.call_args_list[0]
assert first_call == call(mock_ws.send_json(pong))
```